app/models/submission.rb

```diff
@@ -0,0 +1,54 @@
+require 'securerandom'
+
+class Submission < ActiveRecord::Base
+  belongs_to :project
+  has_many :jobs, dependent: :destroy
+
+  store :job_attrs
+
+  attr_accessor :start_frame, :end_frame, :project_dir
+
+  def submit
+    parse_frames
+
+    job = new_job
+    success = job.submit(templated_content)
+
+    success
+  end
+
+  def never_submitted_status
+    'not submitted'
+  end
+
+  private
+
+  def submission_template
+    'jobs/video_jobs/maya_submit.sh.erb'
+  end
+
+  def new_job
```

Comment:
Why do you you choose to create and persist a Job instance before a job is submitted?

Reply:
In osc_machete_rails, the way this was handled was a Job only was created when a batch job was actually submitted. So a workflow (in this case a submission) that could have 1 or more jobs, if it had no jobs, then it would be "not submitted".

Reply:
I think it was a legacy thing where, just in development I wanted to get the id (the database id) to create the directory first. 

Either way, changed to return just a new instance that isn't persisted until after it's submitted.